src/slug.ts

```diff
@@ -0,0 +1,19 @@
+export function slugref(envVar: string): string {
+  return envVar
+    .toLowerCase()
+    .replace(RegExp('refs/(heads|tags)/'), '')
+    .replace(new RegExp('[_/@]', 'g'), '-')
+    .substring(0, 63)
+}
+
+export function slugurl(envVar: string): string {
+  return slugref(envVar).replace(new RegExp('[.]', 'g'), '-')
+}
+
+export function slugurlref(envVar: string): string {
+  return slugurl(slugref(envVar))
+}
+
+export function shortsha(envVar: string): string {
+  return envVar.substring(0, 8)
+}
```

Comment:
Those functions are exported, it might be usefull for next contributors to have a documentation attached to them.

Reply:
I agreed, I have not documented the original version in shell and I already have difficulties to read it again :trollface:

Reply:
fixed